inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -559,9 +559,23 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
         THROW_GNA_LAYER_EXCEPTION(layer) << "Kernel dimensions XY (" << convolution._kernel_x << ", " << convolution._kernel_y << ")"
             << " are bigger than input dimensions WH (" << in_width << "," << in_height << ")";
     }
+    const uint32_t filter_n = convolution._out_depth;
+
     const auto inputs = convolution.insData.front().lock();
     const auto outputs = convolution.outData.front();
 
+    // TODO: questionable why for biases that are not in IR we inventing precision
+    auto biasPrecision = convolution._biases ? convolution._biases->getTensorDesc().getPrecision() : outputs->getPrecision();
+
+    const auto inputPrec = OvGnaTypeIntFromBytes(inputs->getPrecision().size());
+    const auto outputPrec = OvGnaTypeIntFromBytes(outputs->getPrecision().size());
+    const auto weightPrec = OvGnaTypeIntFromBytes(convolution._weights->getTensorDesc().getPrecision().size());
+    const auto biasPrec = OvGnaTypeIntFromBytes(biasPrecision.size());
+
+    cnn2dValidator.ValidateCnn2D(layer->name,
```

Comment:
Not sure that we need a class member. It seems to have functions are enough.

Reply:
IMO object is more generic and  e.g., makes is easy to have separate object for different limitations (i.e., corresponding to other GNA generation).
Is it OK for you?

Reply:
That makes sense, but it not looking generic right now.

Reply:
You are right that it's not perfectly generic right now.
But IMO current approach gives significant potential for future reuse.
Especially that there are issues like 53714 currently being implemented, which can leverage this potential.
IMO going back from a bit more OO desing to a fixed function approach may be a waste of time.

@nkogteva What do you propose then?

Reply:
@kbruniec I see your point. Maybe it should be handled in the scope of another task.